File: 1342.Number_of_Steps_to_Reduce_a_Number_to_Zero/solution.py
def numberOfSteps(num: int) -> int:
    i = 0
    while num > 0:
        if num % 2 == 0:
            num /= 2
        else:
            num -= 1
        i += 1
    return i


print(numberOfSteps(14))
print(numberOfSteps(8))
print(numberOfSteps(123))

File: 1342.Number_of_Steps_to_Reduce_a_Number_to_Zero/solution.js
// Accepted

/**
 * @param {number} num
 * @return {number}
 */
var numberOfSteps = function(num) {
  let i = 0;
  while (num > 0) {
    if (num % 2 === 0) {
      num = num / 2;
    } else {
      num = num - 1;
    }
    i++
  }
  return i;
};

// Case 1
console.log(numberOfSteps(14));

// Case 2
console.log(numberOfSteps(8));

// Case 3
console.log(numberOfSteps(123));